src/solve.jl

```diff
@@ -2,14 +2,8 @@ function SciMLBase.solve(prob::NonlinearProblem,
                          alg::AbstractNonlinearSolveAlgorithm, args...;
                          kwargs...)
   solver = init(prob, alg, args...; kwargs...)
-  sol = solve!(solver)
+  sol, resid = solve!(solver)
```

Comment:
`solve!` has a standard interface that shouldn't be broken. We should just stick the `resid` in the `sol`.

Reply:
Okay, I will update the `sol` struct with `resid`.